src/Common.hpp

```diff
@@ -1,67 +1,101 @@
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of Willow Garage, Inc. nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ ********************************************************************/
 #pragma once
+#include <rclcpp/serialized_message.hpp>
+#include <rclcpp/time.hpp>
 #include <stdexcept>
-#include "ShapeShifterMsg.hpp"
 
 namespace snapshotter
 {
-    enum class BagCompression
-    {
-        NONE,
-        SLOW, //good compression but takes a long time
-        FAST  //a little bit of compression but fast
-    };
+using SerializedMsgPtr = std::shared_ptr<const rclcpp::SerializedMessage>;
 
-    struct BagWriteException : std::runtime_error
-    {
-        using std::runtime_error::runtime_error;
-    };
+static const rclcpp::Time minValidTimeStamp = rclcpp::Time(0, 0, RCL_ROS_TIME);
 
-    struct BufferEntry
-    {
-        /**ATTENTION The BufferEntries are copied when writting to the file.
-         *           This becomes a problem when the BufferEntry grows in size.
-         *           Currently this is fine because it only contains a pointer and two ints.
-         *           DO NOT ADD MORE ATTRIBUTES! (or if you do, rethink how data is written
-         *           to the bag file) */
+enum class BagCompression
+{
+    NONE,
+    SLOW, // good compression but takes a long time
+    FAST // a little bit of compression but fast
+};
+
+struct BagWriteException : std::runtime_error
+{
+    using std::runtime_error::runtime_error;
+};
+
+struct TopicMetadata
+{
+    TopicMetadata(const std::string& topicName, const std::string& topicTypeName, bool transient_local,
+                  uint16_t metaDataIdx) :
+        topicName(topicName),
+        topicTypeName(topicTypeName),
+        transient_local(transient_local),
+        metaDataIdx(metaDataIdx)
+
+    {}
```

Comment:
```suggestion
struct TopicMetadata
{
    TopicMetadata(const std::string& topicName, const std::string& topicTypeName, bool transientLocal,
                  uint16_t metaDataIdx) :
        topicName(topicName),
        topicTypeName(topicTypeName),
        transientLocal(transientLocal),
        metaDataIdx(metaDataIdx)

    {}
```

Reply:
this is self explanatory in the ros2 context